### DISCUSSION TITLE
Sugestões Gerais

### DISCUSSION THREAD

**Discussion by RFS-Solutions** [RamonSilva20/mapos#2606] [Ideas]
Boa tarde,

Seguem algumas sugestões que ach oque poderiam ser implementadas, caso gostem, para melhorar ainda mais o sistema e o controle.

**Status das Ordens de Serviço**
Seria interessante que fosse separado a **Situação de Financeiro** do **Status da OS**. Pois existem casos que a pessoa paga antes da execução do serviço. Ou seja, a OS pode estar Faturada (Paga) e em Execução ou Pendente de Peças, etc... 

**Ordens de Serviço Recorrente**
Afim de poder gerar uma agenda mais precisa para clientes que executamos serviços recorrentes ou até mesmo para já determinar a periodicidade das preventivas.

**Ordens de Serviço Retorno em Garantia**
Quando um equipamento retorna por problemas que a garantia cobre, poderíamos adicionar na OS original para ter um histórico preciso do equipamento em questão.

**Bloqueio de Clientes**
Uma outra sugestão de melhoria seria incluir a opção de "**_Bloqueio_**" do cliente no cadastro. Isso para evitar de se executar um serviço para um cliente devedor.

**TAGs para Identificar Clientes por categoria (Ex. Contrato Bronze, Ouro,etc...)**

Assim como podemos identificar os Forncedores, seria interessante termos a opção de criar TAGs de identificação para clientes que possuem algum tipo de serviço recorrente/contrato.

**21/12/204**

**Anexos**
No caso dos Anexos, poderia existir a possibilidade de identificar oque se está anexando, no caso, por uma legenda.

**OS - Retirada do Equipamento**
Poderia se acrescentar um campo com a data da entrega/retirada do equipamento, que no caso não é o mesmo que a "Data Final" que se refere ao prazo de conclusão prevista.

**22/12/2024**

Vi algo interessante em um sistema antigo, no ato da abertura da OS, era preciso preencher um checklist das condições de entrada do equipamento na assistência. No Checklist poderia ser adicionados textos e fotos referentes ao equipamento.

Essas são apenas algumas sugestões.

Att.

**Comment by sarkozin** [comment]
Adoraria fazer essas melhorias, porem ao que aparenta o projeto não esta aceitando novas features :(

**Reply by willph** [reply]
Voltará a aceitar assim que for lançado a versão do map-os em laravel. Pois tudo está sendo migrado.

Muito obrigado por compartilhar conosco um pouco do seu tempo para melhoria do sistema map-os.

**Reply by RFS-Solutions** [reply]
> laravel

Essa migração, quando liberada, ocorrerá automaticamente pelo sistema?

**Comment by systemilc** [comment]
O sistema está incrível, parabéns aos desenvolvedores. Estou a alguns dias usando e testando o sistema, senti falta de cadastro de categorias no produto e algo relacionado a cadastro de campos extras nas ordens de serviços.

Essa questão dos campos extras, poderia funcionar como no opencart, possui uma tela para cadastrar o campo onde o administrador irá escolher o tipo do campo, nome e posição. dessa forma o sistema poderá se adaptar a qualquer tipo de serviço.

**Comment by reginalopes** [comment]
Adoro o MapOS, já o uso há algum tempo para minha MINI GRÁFICA RÁPIDA. Porém, eu queria adicionar um check-box, para assinalar se é uma DESPESA OU RECEITA RECORRENTE. Espero que na próxima vcs consigam adicionar essa atualização... Eu tentei, mas não consegui.

Algo assim, tanto para despesa quanto para receita

<img width="602" height="517" alt="image" src="https://github.com/user-attachments/assets/f30514b9-98f8-48a0-a18e-c689134c115b" />

Por outro lado, adicionei uma calculadora no backend, que tem me ajudado muito...
![Screenshot_53](https://github.com/user-attachments/assets/382c8858-5cee-4496-80ce-bc6090016f9d)

**Reply by Fesantt** [reply]
kkkkkkkkkkkk
